packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -415,28 +416,27 @@ export const resolvers: GQLResolver = {
         )
       }
     },
-    async fetchRecordDetailsForVerification(_, { id }, { headers }) {
+    async fetchRecordDetailsForVerification(_, { id }, context) {
       try {
         const token = await getAnonymousToken()
-        headers.Authorization = `Bearer ${token}`
+        context.headers.Authorization = `Bearer ${token}`
         const authHeader = {
-          Authorization: headers.Authorization
+          Authorization: context.headers.Authorization
         }
-        const taskEntry = await getTaskEntry(id, authHeader)
+        context.record = await getRecordById(id, authHeader.Authorization)
```

Comment:
does mutating the context do something special or should we just assign the header to its own variable? the variable could be passed directly to `getRecordById` to avoid making the `authHeader` variable which seems unnecessary

Reply:
Context is read in type-resolvers. So I guess, it should be kept with proper values.